Update to python3 and add output dir

kksgandhi opened the pull request:
Updates the script to use python3. Also, all output files now go in a directory of the user's choosing (by default, base/output)

kksgandhi committed:
update script to python3

kksgandhi committed:
fix hashlib bug

strings must be encoded before being passed to hashlib, so when the file
is read, "encode('utf-8')" must be run on the file content string.

kksgandhi committed:
working with bugs

kksgandhi committed:
fix two bugs

first, unnecessary files were being written to the base directory.
Honestly not sure what I did to fix that

second, the program was only updating when the main file was changed.
Now it appropriately changes when any relevant file is changed

kksgandhi committed:
create outputdir folder if it does not exist

kksgandhi committed:
remove texbuild.log file

kksgandhi committed:
update readme

kksgandhi committed:
make README into a markdown file

kksgandhi committed:
update README

kksgandhi committed:
Update README.md

pbiggar commented on README.md:
I think this is left in from the fork?

kksgandhi replied:
Good point, changed the wording somewhat. Let me know if it's still too unprofessional.

pbiggar commented on texbuild:
```diff
@@ -155,17 +150,18 @@ while (True):
 	debug ("copying log")
-	shutil.copy (logfile, "t.log")
-	shutil.copy (auxfile, "t.aux") # for bibtex
+	# shutil.copy (logfile, "t.log")
```
Does something need to happen here?

kksgandhi replied:
Oddly enough, no. 

 - The logfile can stay in output/
 - Bibtex works, even if the aux file isn't copied. Not entirely sure why, but ah well

Regardless, I removed the debug statement since that action is no longer conducted.

pbiggar reviewed (changes requested):
Wow, thanks for the PR on this old repo! Some small questions but generally happy to merge once they're solved.

kksgandhi committed:
update README

kksgandhi committed:
Merge branch 'master' of https://github.com/kksgandhi/texbuild

kksgandhi committed:
remove debug statement about copying log

kksgandhi committed:
update README

kksgandhi commented:
Ok, I've made the requested changes, let me know if there's any more!

kksgandhi commented:
By the way, is there a reason you never updated this repository yourself? Do you use a different build system now, or just don't write in LaTeX anymore?

kksgandhi committed:
don't encode the files, just read them as binary

kksgandhi committed:
convert tabs to spaces

kksgandhi committed:
run bibtex on every aux file

previously this script only ran bibtex on the base aux file. This change
allows for multiple bibliography files

pbiggar commented on texbuild:
```diff
@@ -21,151 +25,151 @@ import sys
-	debug ("copying log")
-	shutil.copy (logfile, "t.log")
-	shutil.copy (auxfile, "t.aux") # for bibtex
+    # debug ("copying log")
```
I guess these lines can be removed?

pbiggar commented on texbuild:
```diff
@@ -21,151 +25,151 @@ import sys
-	print str(time.strftime("%H:%M:%S")) + " - " + message
+    debug (message)
 
+subprocess.call(['mkdir', '-p', outputdir])
```
I think python probably has a built-in, platform-independent version of this?

pbiggar commented on README.md:
Could we rephrase this without the "I"? Maybe "sioyek is a good choice to use with this scripts as..."

kksgandhi committed:
use os mkdir instead of calling shell

kksgandhi committed:
remove unnecessary code comments

kksgandhi committed:
update readme

kksgandhi committed:
slowly ramp up main loop sleep time

if the user is idle, the main loop will be polling every 0.2 seconds.
This change causes the main loop to sleep longer and longer each time
saving marginal system resources

kksgandhi committed:
remove 'msg' function

kksgandhi committed:
give texbuild the python file extension

kksgandhi commented:
Made the changes you mentioned, along with a minor change to how the main loop sleep functions.

pbiggar commented:
Looks great, thanks!

kksgandhi commented:
Awesome, thank you so much!
